packages/lesswrong/components/search/SearchBarResults.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import { registerComponent, Components } from 'meteor/vulcan:core';
+import PropTypes from 'prop-types';
+import {
+  Hits,
+  Configure,
+  Index } from 'react-instantsearch/dom';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = theme => ({
+  root: {
+    backgroundColor: theme.palette.background.default,
+    transition: "opacity .1s ease-in-out",
+    height: 2000,
+    width:"100%",
+    overflow: "hidden",
+    position: "absolute",
+    left: 0,
+    top: 64,
+    [theme.breakpoints.down('small')]: {
+      top:48,
+    }
+  },
+  searchResults: {
+    maxWidth: 1200,
+    margin: "auto",
+  },
+  searchList: {
+    padding:theme.spacing.unit*2
+  },
+  header: {
+    display:"flex",
+    justifyContent:"space-between",
+    alignItems: "center",
+    paddingTop: theme.spacing.unit*2,
+    paddingBottom: theme.spacing.unit*2,
+    paddingLeft: theme.spacing.unit,
+    paddingRight: theme.spacing.unit,
+    '& h1': {
+      margin:0
+    }
+  },
+})
+
+class SearchBarResults extends Component {
+
+  render() {
+    const { classes } = this.props
+
+    return <div className={classes.root}>
+      <Grid container className={classes.searchResults}>
+        <Components.ErrorBoundary>
+          <Grid item xs={12} sm={6} md={5} className={classes.searchList}>
+            <Index indexName="test_posts">
+              <div className={classes.header}>
+                <Typography variant="display1">Posts</Typography>
+                <Components.SearchPagination />
+              </div>
+
+              <Configure hitsPerPage={8} />
+              <Hits hitComponent={(props) => <Components.PostsSearchHit clickAction={this.closeSearch} {...props} />} />
+            </Index>
+          </Grid>
+        </Components.ErrorBoundary>
+        <Components.ErrorBoundary>
+          <Grid item xs={12} sm={6} md={4} className={classes.searchList}>
+            <Index indexName="test_comments">
+              <div className={classes.header}>
+                <Typography variant="display1">Comments</Typography>
+                <Components.SearchPagination />
+              </div>
+              <Configure hitsPerPage={8} />
+              <Hits hitComponent={(props) => <Components.CommentsSearchHit clickAction={this.closeSearch} {...props} />} />
+            </Index>
+          </Grid>
+        </Components.ErrorBoundary>
+        <Components.ErrorBoundary>
```

Comment:
In general I think we kind of go overboard when it comes to error boundaries, but I'm interested to hear your thoughts. How do you go about deciding if there should be an error boundary?

Reply:
I asked Habryka and Jim about this at the time, and the idea was "whenever there's a thing that might break independently while leaving other stuff on the page that that the user would derive value from, use an error boundary." In this case, something might break within the posts, users or comments list, and it'd be nice if the other two didn't break also in that case.